mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultLongIdentifierBridge.java

```diff
@@ -23,4 +23,9 @@ public Long fromDocumentIdentifier(String documentIdentifier, IdentifierBridgeFr
 		return Long.parseLong( documentIdentifier );
 	}
 
+	@Override
+	public boolean isCompatibleWith(IdentifierBridge<?> bridge) {
+		return false;
```

Comment:
Something like return getClass().equals( bridge.getClass() ) would be more appropriate.